adagrams/game.py

```diff
@@ -1,11 +1,273 @@
+        "L": 1,
+        "N": 1,
+        "R": 1,
+        "S": 1,
+        "T": 1,
+        "D": 2,
+        "G": 2,
+        "B": 3,
+        "C": 3,
+        "M": 3,
+        "P": 3,
+        "F": 4,
+        "H": 4,
+        "V": 4,
+        "W": 4,
+        "Y": 4,
+        "K": 5,
+        "J": 8,
+        "X": 8,
+        "Q": 10,
+        "Z": 10
+    }
+    
+    uppercase_word = word.upper()
+    score_count = 0
+    
+    if 7 <= len(uppercase_word) <= 10:
+        score_count += 8
+        
+    for letter in list(uppercase_word):
+        if letter in score_dict:
+            score_count += score_dict[letter]
+    return score_count
+
+
+########     WAVE 4   ################
+######## Helper Functions ###########
+# Get/generate list of tupes for all words in word_list where tuple = ("word", word_score)
+def get_score_list(word_list):
+
+    word_scores = []
+    for i in range(len(word_list)):
+
+        #Calculate score of current word
+        current_word = word_list[i]
+        current_word_score = score_word(current_word)
+
+        # Add tuple of (current word, current word score) to 
+        word_scores.append((current_word, current_word_score))
+
+    return word_scores # returns list of tuples ("word", word score)
+
+
+def get_max_score(word_list):
+    #Constants
+    WORD_INDEX = 0
+    SCORE_INDEX = 1
+    
+    # Call list of tuples from helper function: get_score_list(word_list)
+    all_word_scores = get_score_list(word_list) 
+
+    # Assume the first item's word score in list of tuples is the max-score
+    # Change current_max_score if loop comes across a larger value
+    current_max_score = all_word_scores[WORD_INDEX][SCORE_INDEX]
 
+    for word, score in all_word_scores:
+        # Iterate and check which score is the highest: 
+        if score > current_max_score:
+            current_max_score = score
+            
+        else:
+            continue
+    return current_max_score
+
+
+def get_shortest_len(final_list):
+
+    WORD_IDX = 0
+    shortest_len = len(final_list[0][WORD_IDX])
+
+    for i in range(len(final_list)):
+        if len(final_list[i][WORD_IDX]) < shortest_len:
+            shortest_len = len(final_list[i][WORD_IDX])
+
+    # integer of shortest length of string in final_list
+    return shortest_len 
+
+
+# WAVE 4 FUNCTION - to be tested #
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+
+    highest_score = get_max_score(word_list)
+    all_word_scores = get_score_list(word_list)
+    high_score_words = []
+
+    for current_word, current_score in all_word_scores:
+
+        if current_score == highest_score:
+            high_score_words.append((current_word, current_score))
+
+        else:
+            continue
+    
+    
+    shortest_len = get_shortest_len(high_score_words)
+    high_score_pair = () # return value for function with (word, score)
+
+    for final_word, final_score in high_score_words:
+        if len(high_score_words) == 1:
```

Comment:
This check could come out of the `for..loop` since we wouldn't need to run the loop in this case.